### PR TITLE
uri: deprecate 'yes' or 'no' value in follow_redirects param

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY

* 'yes' or 'no' value deprecated and set to removal in version 2.22

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
- Bugfix Pull Request

### CODE REVIEW

File: lib/ansible/modules/uri.py [hunk 1]
```diff
@@ -107,12 +107,14 @@
...
-        where V(true) is the equivalent of V(all) and V(false) is the equivalent of V(safe). V(true) and V(false)
-        are deprecated and will be removed in some future version of Ansible.
+      - Whether or not the URI module should follow redirects.
+      - V(all) will follow all redirects.
+      - V(safe) will follow only "safe" redirects, where "safe" means that the client is only
+        doing a GET or HEAD on the URI to which it is being redirected.
+      - V(none) will not follow any redirects.
+      - Note that V(true) and V(false) choices are accepted for backwards compatibility,
+        where V(true) is the equivalent of V(all) and V(false) is the equivalent of V(safe).
+        V(true) and V(false) are deprecated and will be removed in Ansible 2.22.
```
Comment (bcoca) [inline lib/ansible/modules/uri.py]: its confusing to use true/false here when choices are explicitly yes/no ... im guessing this was a 'fix' from those 'standarizing' booleans. It should reflect yes/no as the field is a string and not a boolean.

File: lib/ansible/modules/uri.py [hunk 2]
```diff
@@ -107,12 +107,14 @@
...
     type: str
     choices: ['all', 'no', 'none', 'safe', 'urllib2', 'yes']
```
Comment (bcoca) [inline lib/ansible/modules/uri.py]: make choices a dict, keeps the description ON the option and cannot be missed by the user:
```
choices:
   all: Will follow all redirects
   none: no redirects will be followed
   safe: Only redirects doing GET or HEAD requests will be followed
   no: (DEPRECATED, removed in 2.22, use 'none') alias of 'none'
...
```

this would have prevented 'true/false' conversion of the 'yes/no' options above

File: lib/ansible/modules/uri.py [hunk 3]
```diff
@@ -107,15 +107,16 @@
...
-        are deprecated and will be removed in some future version of Ansible.
+      - Whether or not the URI module should follow redirects.
     type: str
-    choices: ['all', 'no', 'none', 'safe', 'urllib2', 'yes']
     default: safe
+    choices:
+      all: Will follow all redirects.
+      none: Will not follow any redirects.
+      safe: Only redirects doing GET or HEAD requests will be followed.
+      urllib2: Automatically follows HTTP redirects.
```
Comment (bcoca) [inline lib/ansible/modules/uri.py]: isn't this one really 'defer to urllib2 behavior (As of writing this follows HTTP redirects)'

File: lib/ansible/modules/uri.py [hunk 4]
```diff
@@ -107,15 +107,16 @@
     default: no
   follow_redirects:
     description:
-      - Whether or not the URI module should follow redirects. V(all) will follow all redirects.
-        V(safe) will follow only "safe" redirects, where "safe" means that the client is only
-        doing a GET or HEAD on the URI to which it is being redirected. V(none) will not follow
-        any redirects. Note that V(true) and V(false) choices are accepted for backwards compatibility,
-        where V(true) is the equivalent of V(all) and V(false) is the equivalent of V(safe). V(true) and V(false)
-        are deprecated and will be removed in some future version of Ansible.
+      - Whether or not the URI module should follow redirects.
```
Comment (bcoca) [inline lib/ansible/modules/uri.py:110]: we should backport/revert the change for true/false in previous versions

Comment (Akasurde) [reply lib/ansible/modules/uri.py:110]: sure.
